### PR TITLE
backing: Remove redundant erasure encoding

### PR DESCRIPTION
**Pull request by sandreim**
In candidate backing we are calling `erasure_coding::obtain_chunks_v1` to get the erasure root and check it vs the expected root, but we drop the result and do this again in `av store`. This PR moves the check in `av store` and removes the redundant one in backing.

### CODE REVIEW

File: node/core/backing/src/lib.rs [hunk 1]
```diff
@@ -593,23 +595,17 @@ async fn make_pov_available(
...
-		if erasure_root != expected_erasure_root {
-			return Ok(Err(InvalidErasureRoot))
-		}
-	}
-
 	{
 		let _span = span.as_ref().map(|s| s.child("store-data").with_candidate(candidate_hash));
 
-		store_available_data(sender, n_validators as u32, candidate_hash, available_data).await?;
+		store_available_data(
```
Comment (rphmeier) [inline node/core/backing/src/lib.rs:601]: Let's add a comment here to explain that this is checking the erasure root

File: node/core/av-store/src/lib.rs [hunk 2]
```diff
@@ -1276,9 +1287,19 @@ fn store_available_data(
 		},
 	};
 
+	let erasure_span = jaeger::Span::new(candidate_hash, "erasure-coding")
+		.with_candidate(candidate_hash)
+		.with_pov(&available_data.pov);
+
```
Comment (rphmeier) [inline node/core/av-store/src/lib.rs:1293]: Now that this is consensus-critical, let's add a comment here that this check is relied upon for candidate validity.

**Review by rphmeier — APPROVED** [review]
LGTM modulo nits. Saving an erasure-encoding seems like a pretty big win.

File: node/core/backing/src/lib.rs (hunk 1 above)
Comment (sandreim) [reply node/core/backing/src/lib.rs]: done

**Comment by sandreim** [comment]
> Does the guide need to be updated?

Updated the guide.

**Review by tdimitrov — APPROVED** [review]
Not an expert at these subsystems but the changes look sane to me.

**Review by alexggh — APPROVED** [review]

**Comment by sandreim** [comment]
bot merge

**Pull request merged**